### ISSUE TITLE
Accessibility Testing for Veteran Facing Forms - Quick Submit

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/va.gov-team#76159]
### Product information

- [X] Team name, product name, and feature name have been added to the title of this issue.
- [X] Team label, product label and feature label (if applicable) have been added to this issue.

### Who completed the use of color and color contrast test?

@jeana-adhoc 

[Main url used for testing](https://preprod.eauth.va.gov/accessva/?cspSelectFor=quicksubmit)

### Use of color and color contrast checks

- [ ] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [ ] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [ ] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
- [ ] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [ ] Any charts, maps, infographics, and tables convey all information without only relying on color
- [ ] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")

### How did color testing go?

**Launch blocking: There are color contrast issues on each page.**  The following list is not all encompassing. 

- [WCAG SC 1.4.3_AA](https://www.w3.org/WAI/WCAG21/quickref/#contrast-minimum)[WCAG SC 1.4.6_AAA](https://www.w3.org/WAI/WCAG21/quickref/#contrast-enhanced)

<img width="99" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/361e49fb-6d25-480f-a880-fbeece22ff20">

- Color contrast: 1.91:1   
- Text Size: Normal   
- WCAG AA: Fail   
- WCAG AAA: Fail   

<img width="193" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/5acd6541-3069-489b-b5a1-4b13735d9ee0">

- Contrast Ratio: 4.47:1
- Text Size: Normal
- WCAG AA: Fail
- WCAG AAA: Fail

<img width="540" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/40a12df5-bb5a-484e-9160-49834d790843">

- Contrast Ratio: 1.38:1
- Text Size: Large
- WCAG AA:Fail
- WCAG AAA:Fail

**Links are not underlined**
<img width="188" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/20a1ee19-d14c-4dc7-830c-26a3e914200d">

### Who completed the axe scans?

@jeana-adhoc 

### axe checks

- [X] Each page has been scanned using axe (results shared below)
- [ ] axe is integrated into your end-to-end testing - N/A - We did not build these pages.

### axe DevTools scan results

| Axe Results | Page |
| -- | -- |
| 5 issues (Color contrast) | [Register](https://staging.digitization.gcio.com/va/upload/register) |
| 1 issue (Color contrast) | [Landing page](https://staging.digitization.gcio.com/va/upload/home) |
| 2 issues (Color contrast & Form labels) | [Upload page](https://staging.digitization.gcio.com/va/upload/uploads) |
| 4 Issues (Color contrast, List type, Aria, Headings | [After a file is uploaded](https://staging.digitization.gcio.com/va/upload/home) |

### How did axe testing go?

There were issues on each page.

Axe identified the color contrast issues, and identified a form input that did not have a label.

After a file is uploaded, Axe identified further issues with markup including aria, headings, and lists.

### Who completed the content zoom and reflow test?

@jeana-adhoc 

### Content zoom and reflow checks

- [X] All page elements are readable and usable at 200% zoom
- [X] All page elements are readable and usable at 300% zoom
- [X] All page elements are readable and usable at 400% zoom

### How did content zoom and reflow testing go?

Was unable to test zoom on the registration page. But the upload and langing pages were fine with no issues.

### Who completed the keyboard navigation test?

@jeana-adhoc 

### Keyboard navigation checks

- [X] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [ ] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [ ] All elements under focus have a visible focus indicator
- [X] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

**Launch blocking: Item is focusable but does not have accessible label read out** (Not an exhaustive list):

The application pages show 3 "tool tips" that when you click on them with a mouse, they show more information on the page. However, these options are *not* keyboard accessible and do not respond to the expected, enter, spacebar, or down arrow key to activate. See: [WCAG SC 2.1.1_A](https://www.w3.org/WAI/WCAG21/quickref/#keyboard) - 
Page functionality isn't keyboard accessible. |  

https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/d9e532f2-4d82-4c2c-a156-165ef654009d

- <img width="350" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/00133f1f-59cf-4698-9099-661965ce4949">
- <img width="350" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/20513876-34ca-4408-bd90-7d9ebdea299c">
- <img width="350" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/850cd6af-a5a0-4974-bfea-ba9a7ceab3ba">

**Focus state of quick submit button is not visible**   
Not focused:   
<img width="339" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/51b4def7-0248-4a0f-b8c0-bb872db4ca16">

Focused:
<img width="324" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/3ce80518-614b-427d-95ee-5dc1241f94c7">

The focus state of the QuickSubmit button is not noticeable. There is a color change from `#005ea2` to `#004b82`

#### Launch Blocking: No keyboard accessible form fields
[WCAG SC 3.3.2_A](https://www.w3.org/WAI/WCAG21/quickref/#labels-or-instructions)

For new users who have never used quick submit before, they have to select their user type. This is the screen they get.
- The 'radio buttons' are not radio buttons at all. 
- This is not a form 
- Keyboard users cannot navigate into the 'form' and make selections at all.

https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/44085624-d2a5-4af3-8763-bcf96cd17f9e

#### Launch blocking: Labels or instructions aren't provided when content requires user input.

[WCAG SC 3.3.2_A](https://www.w3.org/WAI/WCAG21/quickref/#labels-or-instructions)

The upload page has a missing form labels so that when you tab into a field, a non-sighted user would not hear the purpose of the field announced 

https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/77070791-fa85-4c1d-9a1a-d04da252a311

#### Content, entry points, and link labels don't use plain language or familiar words, terms, and acronyms that users understand. 
[WCAG SC 3.1.4 AAA](https://www.w3.org/TR/UNDERSTANDING-WCAG20/meaning-located.html)
The field labeled "MI" is not very clear in indicating what information is needed here. A screen reader announces the letters "M" "I". (See video above)

#### Launch blocking: Link or button is not working as expected or did not navigate to the intended destination.
The large quick submit button is used as a navigation/link element to the upload page. It is not actually doing anything or changing a state.
<img width="341" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/8195c2f5-902c-419b-a2a8-6ed596ed3a6c">

#### Launch blocking: Name and role can't be programmatically determined for UI components.  
[WCAG SC 4.1.2 A](https://www.w3.org/WAI/WCAG22/quickref/?versions=2.1&showtechniques=412#name-role-value)

https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/21875ea9-c3ff-4972-8108-3aa5366e9453

Once a file is uploaded, the user is presented with icons next to the file name. A sighted user sees an eye icon and a trashcan icon. When navigating to them using a screen reader, the screen reader only announces "Link". It does not state the purpose of the link. Mac Voiceover does not announce `title`'s defined in the `a href` tag.

#### Launch blocking: Content causes the user's keyboard focus to get trapped   
[WCAG SC 2.1.2_A](https://www.w3.org/WAI/WCAG21/quickref/#no-keyboard-trap)  

https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/25bf861a-479d-4d82-8f6f-513fe4416c63

When navigating the preview or delete options using the keyboard only, once the links are pressed and the modal appears, focus never goes into the modal, the keyboard user stays trapped behind the modal. In the video, I did use the mouse to click into the modal to see if I could then navigate around within the modal, and I could to an extent. I could never use the keyboard to close the modal. This means a screen reader user would never be able to go into the modal to explore more.

#### Tabular data should be represented in a `<table>` 
The tables are coded as divs with many aria attributes applied. `<table>` code can be used correctly to make a more accessible table; table should include a `<caption>` to help describe the table to screen reader users.

#### Required Label
For users using zoom text or screen mangification, they may not see the "required" label floating to the right above the text input. Consider aligning it left with the content. The required label is read by the screen reader however (this is good).   

<img width="313" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/7ada5e73-e62a-43fd-a610-e309e2fdce09">

#### Your page doesn't follow heading hierarchy. Headings aren't properly nested, and headings and/or labels aren't relevant to the page. 

[WCAG SC 1.3.1_A](https://dsvavsp.testrail.io/index.php?/cases/view/5165)[WCAG SC 2.4.6_AA](https://dsvavsp.testrail.io/index.php?/cases/view/5157)

<img width="650" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/73244f33-bdad-4b86-8875-c3bad14ea9e3">

Once a file is uploaded, the `Upload Success` is coded as an `<h5>` and it should be an `<h3>`

#### Alert and status messages aren't announced without receiving focus.

<img width="650" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/73244f33-bdad-4b86-8875-c3bad14ea9e3">

The new card that appears after upload, is improperly coded with a `role="alert"` on an <li>. Also, the `x` is focusable in the card, but I cannot do anything with the keyboard to 'click' the "x" as the focus indicates I should be able to.

### Final note & Recommendation
This is not a full exhaustive audit. This is a summary of the many launch blocking issues that render the application unusable to screen reader users.  After these issues are remediated, the tool would need another round of auditing/testing to see if we can use this for usability research with screen readers.

**Comment by jeana-adhoc** [comment]
I will finish this in the next sprint.

**Comment by jeana-adhoc** [comment]
In the write-up above, I've identified at least 8 launch blocking issues. This is not an exhaustive audit for several reasons:

1. Only one user path was tested (family member)
2. Not all states were tested
3. Many bugs identified were repeated throughout the application and not every instance was documented

As is, a screen reader user cannot do the following:
1. Select their user type upon account setup (They'd never be able to get past the first screen).
2. If they were able to get help and get past the first screen a screen reader user would never know what types of files they can upload (for example, txt files are not accepted, but you don't know that unless you read the "tool tip" which a screen reader cannot access).
3. Screen reader users cannot review the documents they uploaded because the modal is not focusable (if they were to open the modal, they would be stuck, and the only way out would be to refresh the page or leave the application).

**Comment by artsymartha68** [comment]
I opened a request for a 508 audit today.

**Comment by jeana-adhoc** [comment]
[Presentation link for quick submit findings](https://docs.google.com/presentation/d/1_FIfYmRW-tC35fswecSe9D2pamZSWNa_b3gRYojwCQQ/edit?usp=sharing)
